### DISCUSSION TITLE
How does the $transaction method handle update failures?

### DISCUSSION THREAD

**Discussion by seanemmer** [prisma/prisma#4275] [Q&A] [answered]
The `update` method throws an error if the entry is not found in the database.  What happens if an update happens inside of a transaction?  Does the entire transaction fail to commit?  Is there any way to allow the transaction to tolerate an update failure?

**Comment by ryands17** [comment, marked as answer]
@seanemmer 👋 

Yes the entire transaction will fail to commit if the record is not found.

> Is there any way to allow the transaction to tolerate an update failure?
Although not ideal, you can use `updateMany` to tolerate a failed update for the timebeing. I would suggest creating a request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for something like `updateIfExists`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
